src/agent/src/storage/mod.rs

```diff
@@ -55,6 +54,86 @@ pub struct StorageContext<'a> {
     sandbox: &'a Arc<Mutex<Sandbox>>,
 }
 
+/// An implementation of generic storage device.
+pub struct StorageDeviceGeneric {
+    path: String,
+    initialized: bool,
+}
+
+impl std::fmt::Debug for StorageDeviceGeneric {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("StorageDeviceGeneric")
+            .field("path", &self.path)
+            .finish()
+    }
+}
+
+impl Default for StorageDeviceGeneric {
+    fn default() -> Self {
+        StorageDeviceGeneric {
+            path: "".to_string(),
+            initialized: false,
+        }
+    }
+}
+
+impl StorageDeviceGeneric {
+    /// Create a new instance of `StorageStateCommon`.
+    pub fn new(path: String) -> Self {
+        StorageDeviceGeneric {
+            path,
+            initialized: true,
+        }
+    }
+}
+
+impl StorageDevice for StorageDeviceGeneric {
+    fn path(&self) -> &str {
+        &self.path
+    }
+
+    fn cleanup(&self) -> Result<()> {
+        if !self.initialized {
+            return Ok(());
+        } else if self.path.is_empty() {
+            return Err(anyhow!("mountpoint path is empty"));
```

Comment:
If the path is empty, there is no need to clean it up, maybe return `ok()` is better.

Reply:
Fixed